doc/releases/changelog-dev.md

```diff
@@ -24,6 +24,9 @@
 
 <h3>Deprecations 👋</h3>
 
+* Gradient transforms with Jax do not support `argnum` anymore,  `argnums` needs to be used.
+  [(#4076)](https://github.com/PennyLaneAI/pennylane/pull/4076)
+
```

Comment:
Lets put this under "breaking changes" instead because it is the completion of a deprecation cycle and is breaking things, rather than a new deprecation.

Reply:
Ok I agree